nexus/db-queries/src/db/datastore/db_metadata.rs

```diff
@@ -744,4 +864,103 @@ mod test {
         crdb.cleanup().await.unwrap();
         logctx.cleanup_successful();
     }
+
+    #[tokio::test]
+    async fn schema_version_subcomponents_save_progress() {
```

Comment:
Does the `versions_have_idempotent_up` test need to be updated to test by file rather than by directory?

Reply:
I think it's still okay as-is -- that particular test, as well as some of the other schema tests, are "testing the schema changes" more than they are "testing the mechanism to apply schema changes".

I say this referencing the following block of code:

https://github.com/oxidecomputer/omicron/blob/a655ff2a26b1c58335d4ecc17e8263df73045063/nexus/tests/integration_tests/schema.rs#L141-L150

Which appears to already be re-running each individual step, rather than "the whole directory at once".

Reply:
Got it, thanks!

Reply:
Although maybe that loop needs to be inverted? Seems like we want to apply each step `times_to_apply` times, not the whole set....

Reply:
I updated this in 36fdb610b201a4195ae36740eea9edc2cdd1d56b - I think you're right, the test was checking the wrong thing before, and should be fixed now.

Reply:
Okay, this requires more elaboration. When I updated this test, I noticed that it actually started failing.

Specifically, this version threw an error:

https://github.com/oxidecomputer/omicron/blob/a655ff2a26b1c58335d4ecc17e8263df73045063/schema/crdb/10.0.0/up05.sql#L1

As, once the value is dropped, it no longer exists.

Postgres supports syntax to ALTER TYPE DROP VALUE IF EXISTS: https://www.postgresql.org/docs/current/sql-altertype.html

But CockroachDB does not: https://www.cockroachlabs.com/docs/stable/alter-type

To mitigate:
- I have filed a bug against CockroachDB as a feature request: https://github.com/cockroachdb/cockroach/issues/120801
- In 28e6876910f3395e2109ebac53885ad8fdfcf182 , I make an exemption for this schema version